services/worker/src/worker/executor.py

```
@@ -0,0 +1,97 @@
+        self.app_config = app_config
+        self.job_runner_factory = job_runner_factory
+
+    def _create_worker_loop_executor(self) -> OutputExecutor:
+        banner = self.app_config.worker.state_path
+        if not banner:
+            raise ValueError("Failed to create the executor because WORKER_STATE_PATH is missing.")
+        start_worker_loop_command = [
+            sys.executable,
+            START_WORKER_LOOP_PATH,
+            "--print-worker-state-path",
+        ]
+        return OutputExecutor(start_worker_loop_command, banner, timeout=10)
+
+    def start(self) -> None:
+        worker_loop_executor = self._create_worker_loop_executor()
+        worker_loop_executor.start()  # blocking until the banner is printed
+
+        loop = asyncio.get_event_loop()
+        logging.info("Starting heartbeat.")
+        loop.create_task(every(self.heartbeat, seconds=self.app_config.worker.heartbeat_interval_seconds))
+        loop.create_task(every(self.kill_zombies, seconds=self.app_config.worker.kill_zombies_interval_seconds))
+        loop.run_until_complete(
+            every(self.is_worker_alive, worker_loop_executor=worker_loop_executor, seconds=1, stop_on=False)
+        )
+
+    def get_state(self) -> WorkerState:
+        worker_state_path = self.app_config.worker.state_path
+        if not worker_state_path:
+            raise ValueError("Failed to get worker state because WORKER_STATE_PATH is missing.")
+        if os.path.exists(worker_state_path):
+            with FileLock(worker_state_path + ".lock"):
+                try:
+                    with open(worker_state_path, "r") as worker_state_f:
+                        worker_state = json.load(worker_state_f)
+                        return WorkerState(current_job_info=worker_state.get("current_job_info"))
+                except json.JSONDecodeError:
+                    return WorkerState(current_job_info=None)
```

Can we raise an exception here instead of hiding a potential issue?